src/Compilers/CSharp/Portable/BoundTree/BoundIsPatternExpression.cs

```diff
@@ -13,11 +13,13 @@ public BoundDecisionDag GetDecisionDagForLowering(CSharpCompilation compilation)
             BoundDecisionDag decisionDag = this.ReachabilityDecisionDag;
             if (decisionDag.ContainsAnySynthesizedNodes())
             {
+                bool negated = this.Pattern.IsNegated(out var innerPattern);
+                Debug.Assert(negated == this.IsNegated);
                 decisionDag = DecisionDagBuilder.CreateDecisionDagForIsPattern(
                     compilation,
                     this.Syntax,
                     this.Expression,
-                    this.Pattern,
+                    innerPattern,
```

Comment:
>innerPattern [](http://example.com/codeflow?start=20&length=12)

It is not clear why stripping `not` here is the right thing to do. Is there some requirement that this API, ```GetDecisionDagForLowering```, supposed to do this? There is no hint about this on the API. Why nothing special is done for the code path that doesn't enter this ```if```. Perhaps the negation should be handled somewhere else?

Reply:
Ok, I see that we perform the same digging when we create ReachabilityDecisionDag during initial binding and here we should follow the same pattern.